### ISSUE TITLE
errors thrown by `watch_manifest` when adding packages

### ISSUE THREAD

**Issue by ExpandingMan** [timholy/Revise.jl#312]
I still can't figure out under what circumstances this is thrown, but I see it sometimes after adding a package with `]add`.
```julia
ArgumentError: Package CodeTracking not found in current path:
- Run `import Pkg; Pkg.add("CodeTracking")` to install the CodeTracking package.
(v1.1) pkg> 
Stacktrace:
 [1] require(::Module, ::Symbol) at ./loading.jl:823
 [2] eval at ./boot.jl:328 [inlined]
 [3] prepare_thunk(::Module, ::Expr, ::Bool) at /home/msavas200/.julia/packages/JuliaInterpreter/6LimI/src/construct.jl:341
 [4] prepare_thunk(::Module, ::Expr, ::Bool) at /home/msavas200/.julia/packages/JuliaInterpreter/6LimI/src/construct.jl:347
 [5] prepare_thunk at /home/msavas200/.julia/packages/JuliaInterpreter/6LimI/src/construct.jl:333 [inlined]
 [6] #methods_by_execution!#9(::Base.Iterators.Pairs{Symbol,Bool,Tuple{Symbol},NamedTuple{(:define,),Tuple{Bool}}}, ::Function, ::Any, ::Revise.CodeTrackingMethodInfo, ::Dict{Module,Array{Expr,1}}, ::Module, ::Expr) at /home/msavas200/.julia/packages/Revise/UtBAC/src/lowered.jl:47
 [7] #methods_by_execution! at ./none:0 [inlined]
 [8] #eval_with_signatures#59(::Bool, ::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}, ::Function, ::Module, ::Expr) at /home/msavas200/.julia/packages/Revise/UtBAC/src/Revise.jl:344
 [9] #eval_with_signatures at /home/msavas200/.julia/packages/Revise/UtBAC/src/Revise.jl:0 [inlined]
 [10] #instantiate_sigs!#60(::Bool, ::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}, ::Function, ::OrderedCollections.OrderedDict{Module,OrderedCollections.OrderedDict{Revise.RelocatableExpr,Union{Nothing, Array{Any,1}}}}) at /home/msavas200/.julia/packages/Revise/UtBAC/src/Revise.jl:352
 [11] instantiate_sigs! at /home/msavas200/.julia/packages/Revise/UtBAC/src/Revise.jl:349 [inlined]
 [12] maybe_parse_from_cache!(::Revise.PkgData, ::String) at /home/msavas200/.julia/packages/Revise/UtBAC/src/pkgs.jl:224
 [13] (::getfield(Revise, Symbol("##26#27")){String})() at /home/msavas200/.julia/packages/Revise/UtBAC/src/pkgs.jl:370
 [14] with_logstate(::getfield(Revise, Symbol("##26#27")){String}, ::Base.CoreLogging.LogState) at ./logging.jl:395
 [15] with_logger at ./logging.jl:491 [inlined]
 [16] watch_manifest(::String) at /home/msavas200/.julia/packages/Revise/UtBAC/src/pkgs.jl:349
 [17] (::Revise.Rescheduler{typeof(Revise.watch_manifest),Tuple{String}})() at /home/msavas200/.julia/packages/Revise/UtBAC/src/types.jl:235
while evaluating
using CodeTracking
in module Main.JuliaInterpreter
UndefVarError: _precompile_ not defined
Stacktrace:
 [1] macro expansion at /home/msavas200/.julia/packages/JuliaInterpreter/6LimI/src/interpret.jl:64 [inlined]
 [2] #methods_by_execution!#14(::Bool, ::Bool, ::Function, ::Any, ::Revise.CodeTrackingMethodInfo, ::Dict{Module,Array{Expr,1}}, ::JuliaInterpreter.Frame) at /home/msavas200/.julia/packages/Revise/UtBAC/src/lowered.jl:125
 [3] #methods_by_execution! at ./none:0 [inlined]
 [4] #methods_by_execution!#9(::Base.Iterators.Pairs{Symbol,Bool,Tuple{Symbol},NamedTuple{(:define,),Tuple{Bool}}}, ::Function, ::Any, ::Revise.CodeTrackingMethodInfo, ::Dict{Module,Array{Expr,1}}, ::Module, ::Expr) at /home/msavas200/.julia/packages/Revise/UtBAC/src/lowered.jl:49
 [5] #methods_by_execution! at ./none:0 [inlined]
 [6] #eval_with_signatures#59(::Bool, ::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}, ::Function, ::Module, ::Expr) at /home/msavas200/.julia/packages/Revise/UtBAC/src/Revise.jl:344
 [7] #eval_with_signatures at /home/msavas200/.julia/packages/Revise/UtBAC/src/Revise.jl:0 [inlined]
 [8] #instantiate_sigs!#60(::Bool, ::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}, ::Function, ::OrderedCollections.OrderedDict{Module,OrderedCollections.OrderedDict{Revise.RelocatableExpr,Union{Nothing, Array{Any,1}}}}) at /home/msavas200/.julia/packages/Revise/UtBAC/src/Revise.jl:352
 [9] instantiate_sigs! at /home/msavas200/.julia/packages/Revise/UtBAC/src/Revise.jl:349 [inlined]
 [10] maybe_parse_from_cache!(::Revise.PkgData, ::String) at /home/msavas200/.julia/packages/Revise/UtBAC/src/pkgs.jl:224
 [11] (::getfield(Revise, Symbol("##26#27")){String})() at /home/msavas200/.julia/packages/Revise/UtBAC/src/pkgs.jl:370
 [12] with_logstate(::getfield(Revise, Symbol("##26#27")){String}, ::Base.CoreLogging.LogState) at ./logging.jl:395
 [13] with_logger at ./logging.jl:491 [inlined]
 [14] watch_manifest(::String) at /home/msavas200/.julia/packages/Revise/UtBAC/src/pkgs.jl:349
 [15] (::Revise.Rescheduler{typeof(Revise.watch_manifest),Tuple{String}})() at /home/msavas200/.julia/packages/Revise/UtBAC/src/types.jl:235
while evaluating
_precompile_()
in module Main.JuliaInterpreter
```

**Comment by timholy** [comment]
I've not seen this, but of course I long ago needed to `]add CodeTracking`. Anyone else seen this?

**Comment by timholy** [comment]
If you want me to debug this I'll need log files: https://timholy.github.io/Revise.jl/dev/debugging/

**Comment by ExpandingMan** [comment]
I tried doing `]rm JLD2` and then `]add JLD2` with debug logging, but could not reproduce the error.  It's a bit of a heisenbug.  If I figure out how to reproduce it, I'll give you a debug log.

**Comment by Pbellive** [comment]
I saw this for the first time yesterday but I'm not able to reproduce it now. I updated a bunch of packages, then ran a script and saw this error after the script successfully completed. I'm sorry that I can't produce any useful log information but I thought it would be worth adding my report here.

**Comment by KristofferC** [comment]
What are the versions of the relevant packages?

**Comment by fredrikekre** [comment]
I just got this error. I was running one Juila session with Revise, while from another session I updated packages, in particular JuliaInterpreter was updated from 0.5.1 to 0.5.2.

**Comment by timholy** [comment]
For everyone who is getting this, is it specifically about `CodeTracking` being used in `Main.JuliaInterpreter`? Or does it happen with other combinations?

What confuses me is that if I start with a "clean" environment in which I've said `add JuliaInterpreter` but not `add CodeTracking`,
```julia
julia> using JuliaInterpreter

julia> Core.eval(JuliaInterpreter, :(using CodeTracking))

julia> Core.eval(Main.JuliaInterpreter, :(using CodeTracking))

julia> using CodeTracking
ERROR: ArgumentError: Package CodeTracking not found in current path:
- Run `import Pkg; Pkg.add("CodeTracking")` to install the CodeTracking package.

Stacktrace:
 [1] require(::Module, ::Symbol) at ./loading.jl:823
```
which is exactly what I'd expect.

**Comment by ExpandingMan** [comment]
I had *not* explicitly called `add CodeTracking` when I had this error, I believe it was an indirect dependency.

**Comment by timholy** [comment]
Understood. I just meant that when people chime in saying they got the same error, I'm assuming they are not claiming they have the *exact* same stacktrace. (It would be unlikely for others to have a stacktrace starting with `/home/msavas200` :smile:.) I'm just wanting to check that they mean the same CodeTracking/JuliaInterpreter pair, or whether this happens for other pairs too.

**Comment by ExpandingMan** [comment]
Ah, I think I've almost figured it out.  It looks like this is caused any time you talk to the package manager and `Revise` itself is changed somehow, either updated, downgraded whatever.

See if this reproduces it:
First, make sure everything is completely up to date with `up` (and if you have `Revise` on `dev` do `rm Revise` first, then add it back).  Next do `]add Revise@2.1.4`.  This should reproduce the error.

If you can't get it, I'll get you a debug log when I get a chance.

**Comment by timholy** [comment]
Yes, that replicates it for me! I can't promise I'll get to this quickly but it's great to have a MWE. Thanks!

**Comment by timholy** [comment]
OK, I think I figured this out. It turned out to be a module hygiene issue. `Main.JuliaInterpreter` shouldn't actually exist under these circumstances, since you haven't said `using JuliaInterpeter`. It ended up creating 2 JuliaInterpreter modules, a new one under `Main` and the original under `Revise`. Should be fixed in #346.
